iceoryx_posh/include/iceoryx_posh/internal/mepoo/segment_manager.hpp

```diff
@@ -78,21 +79,21 @@ class SegmentManager
 
     struct SegmentUserInformation
     {
-        MemoryManager* m_memoryManager;
+        cxx::optional<MemoryManager*> m_memoryManager;
```

Comment:
I completely agree with your intention here but this solution causes even more problems. Since now you have two possible ways to invalidate memory manager:
1. cxx::nullopt
2. valid optional which contains a nullptr.

I recommand to use here a `cxx::optional<std::reference_wrapper<MemoryManager>>` since a reference can never be null and we can be sure that it always contains a valid value. Otherwise we have to double check everything with this update: one check for optional contains a value and one check for optional value is not nullptr.

Reply:
Good point, done.